### DISCUSSION TITLE
logtail_check_point文件数据时有时无

### DISCUSSION THREAD

**Discussion by leizhou666** [alibaba/ilogtail#981] [Help] [answered]
现象如下：
pod有20+收集配置
![image](https://github.com/alibaba/ilogtail/assets/97587488/e181c819-5711-4e4e-b5a3-7bd63fd7fa36)

正常情况下checkpoint文件如下：
![image](https://github.com/alibaba/ilogtail/assets/97587488/383624c6-9511-409a-a1f9-c923c1f26eaa)

但是间歇性会出现如下异常情况：
![image](https://github.com/alibaba/ilogtail/assets/97587488/6486b6c1-36e0-4cc2-9f25-93c64c6dacf5)
![image](https://github.com/alibaba/ilogtail/assets/97587488/d7b78009-5e85-444e-b16b-6b9784259bbc)

同时：
收集配置check_point_dump_interval时间是300秒，五分钟一次，但是上图里的checkpoint时间竟然只间隔了2分钟
![image](https://github.com/alibaba/ilogtail/assets/97587488/f9f8f5f5-6aa7-4041-9047-30e471ba7d9a)
启动日志里面的配置也是5分钟
![image](https://github.com/alibaba/ilogtail/assets/97587488/55061bd4-b9b6-4f76-bc3b-b96690a9ae57)

**Comment by leizhou666** [comment]
麻烦大佬们帮忙看下这个问题，感谢！

**Comment by leizhou666** [comment]
还有的其他信息是，我重新指向了checkpoint文件的位置为/app/logs/ilogtail/checkpoint/logtail_check_point
而部署的ilogtail的目录为/usr/local/ilogtail

**Comment by linrunqi08** [comment]
这个是在容器里还是在主机上？

**Reply by leizhou666** [reply]
容器上

**Reply by linrunqi08** [reply]
logtail的deployment贴一下

**Reply by leizhou666** [reply]
apiVersion:&nbsp;apps/v1
kind:&nbsp;DaemonSet
metadata:
&nbsp;&nbsp;name:&nbsp;log-collector
&nbsp;&nbsp;namespace:&nbsp;ilogtail
&nbsp;&nbsp;labels:
&nbsp;&nbsp;&nbsp;&nbsp;k8s-app:&nbsp;log-collector
&nbsp;&nbsp;&nbsp;&nbsp;app-name:&nbsp;log-collector
spec:
&nbsp;&nbsp;selector:
&nbsp;&nbsp;&nbsp;&nbsp;matchLabels:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;name:&nbsp;log-collector
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;app-name:&nbsp;log-collector
&nbsp;&nbsp;template:
&nbsp;&nbsp;&nbsp;&nbsp;metadata:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;annotations:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;prometheus.io/port:&nbsp;"9249"
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;prometheus.io/scrape:&nbsp;"true"
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;labels:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;name:&nbsp;log-collector
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;app-name:&nbsp;log-collector
&nbsp;&nbsp;&nbsp;&nbsp;spec:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;volumes:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;name:&nbsp;run
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;hostPath:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;path:&nbsp;/var/run
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;type:&nbsp;Directory
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;name:&nbsp;root
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;hostPath:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;path:&nbsp;/
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;type:&nbsp;Directory
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;name:&nbsp;ilogtail-log
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;hostPath:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;path:&nbsp;/data/logs
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;name:&nbsp;ilog-agent-logs
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;hostPath:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;path:&nbsp;/data/logs/ilogtail
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;type:&nbsp;DirectoryOrCreate
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;name:&nbsp;ilogtail-user-config
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;emptyDir:&nbsp;{}
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;containers:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;name:&nbsp;ilogtail
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;env:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;name:&nbsp;ALIYUN_LOG_ENV_TAGS&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;#&nbsp;add&nbsp;log&nbsp;tags&nbsp;from&nbsp;env
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;value:&nbsp;_node_name_|_node_ip_
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;name:&nbsp;_node_name_
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;valueFrom:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;fieldRef:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;apiVersion:&nbsp;v1
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;fieldPath:&nbsp;spec.nodeName
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;name:&nbsp;_node_ip_
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;valueFrom:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;fieldRef:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;apiVersion:&nbsp;v1
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;fieldPath:&nbsp;status.hostIP
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;name:&nbsp;ALIYUN_LOGTAIL_USER_DEFINED_ID
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;value:&nbsp;"default"
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;name:&nbsp;cpu_usage_limit
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;value:&nbsp;"2"
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;name:&nbsp;mem_usage_limit
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;value:&nbsp;"1800"
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;name:&nbsp;max_read_buffer_size
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;value:&nbsp;"1048576"
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;image:&nbsp;docker.xxxxxx.com/infra/bigdata/rt/ilogtail/ilogtail-1.6.0:v0.0.2
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;resources:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;limits:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;cpu:&nbsp;2000m
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;memory:&nbsp;2Gi
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;nvidia.com/gpu:&nbsp;0
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;requests:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;cpu:&nbsp;500m
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;memory:&nbsp;1Gi
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;nvidia.com/gpu:&nbsp;0
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;ports:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;containerPort:&nbsp;49999
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;volumeMounts:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;name:&nbsp;run
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;mountPath:&nbsp;/var/run
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;name:&nbsp;root
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;mountPath:&nbsp;/logtail_host&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;#&nbsp;for&nbsp;log&nbsp;access&nbsp;on&nbsp;the&nbsp;node
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;mountPropagation:&nbsp;HostToContainer
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;readOnly:&nbsp;true
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;name:&nbsp;ilogtail-log
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;mountPath:&nbsp;/app/logs/
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;name:&nbsp;ilogtail-user-config
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;mountPath:&nbsp;/usr/local/ilogtail/user_yaml_config.d
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;name:&nbsp;log-agent
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;volumeMounts:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;mountPath:&nbsp;/app/logs/ilogtail
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;name:&nbsp;ilog-agent-logs
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;name:&nbsp;ilogtail-user-config
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;mountPath:&nbsp;/app/pipeline
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;args:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;'-n'
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;'ilogtail'
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;'-e'
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;'test'
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;'-c'
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;'default'
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;'-d'
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;'log-collector'
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;'-h'
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;$(NodeName)
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;env:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;name:&nbsp;NodeName
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;valueFrom:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;fieldRef:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;apiVersion:&nbsp;v1
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;fieldPath:&nbsp;spec.nodeName
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;name:&nbsp;PodName
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;valueFrom:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;fieldRef:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;apiVersion:&nbsp;v1
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;fieldPath:&nbsp;metadata.name
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;image:&nbsp;docker.xxxxxx.com/infra/bigdata/rt/ilogtail/log-agent:v0.0.4
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;imagePullPolicy:&nbsp;IfNotPresent
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;resources:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;limits:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;cpu:&nbsp;1000m
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;memory:&nbsp;2Gi
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;requests:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;cpu:&nbsp;200m
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;memory:&nbsp;1Gi

**Reply by linrunqi08** [reply]
采集配置，和日志文件ilogtail.LOG麻烦也提供一下吧

**Comment by leizhou666** [comment]
采集配置其中之一：其他配置都差不多
enable:&nbsp;true
inputs:
-&nbsp;Type:&nbsp;"file_log"
&nbsp;&nbsp;LogPath:&nbsp;"/app/logs/aeolus"
&nbsp;&nbsp;FilePattern:&nbsp;"serverlog.*.log"
&nbsp;&nbsp;ContainerFile:&nbsp;true
&nbsp;&nbsp;ContainerInfo:
&nbsp;&nbsp;&nbsp;&nbsp;K8sNamespaceRegex:&nbsp;"develop"
&nbsp;&nbsp;&nbsp;&nbsp;K8sPodRegex:&nbsp;"^(aeolus.*)$"
processors:
-&nbsp;Type:&nbsp;"processor_split_log_regex"
&nbsp;&nbsp;SplitRegex:&nbsp;"^\\d+-\\d+-\\d+.\\d+:\\d+:\\d+.*"
&nbsp;&nbsp;SplitKey:&nbsp;"content"
&nbsp;&nbsp;PreserveOthers:&nbsp;true
flushers:
-&nbsp;Type:&nbsp;"flusher_kafka_v2"
&nbsp;&nbsp;Brokers:
&nbsp;&nbsp;-&nbsp;"xxxxx1.bj.baidu.internal:9092"
&nbsp;&nbsp;-&nbsp;"xxxxx2.bj.baidu.internal:9092"
&nbsp;&nbsp;-&nbsp;"xxxxx3.bj.baidu.internal:9092"
&nbsp;&nbsp;Authentication:
&nbsp;&nbsp;&nbsp;&nbsp;Kerberos:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;ServiceName:&nbsp;"kafka"
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Realm:&nbsp;"BJ.BAIDU.COM"
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;UseKeyTab:&nbsp;true
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Username:&nbsp;"jd-ilogtail"
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;ConfigPath:&nbsp;"/usr/local/kerberos/krb5.test.conf"
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;KeyTabPath:&nbsp;"/usr/local/kerberos/jd-ilogtail.test.keytab"
&nbsp;&nbsp;Version:&nbsp;"2.8.1"
&nbsp;&nbsp;Topic:&nbsp;"bigdata-rt-collector-aeolus-serverlog-test"
&nbsp;&nbsp;MaxMessageBytes:&nbsp;20971520

**Comment by leizhou666** [comment]
apsara_log_conf.json：
{
	"Loggers" : 
	{
		"/" : 
		{
			"AsyncFileSink" : "WARNING"
		},
		"/apsara/sls/ilogtail" : 
		{
			"AsyncFileSink" : "INFO"
		},
		"/apsara/sls/ilogtail/profile" : 
		{
			"AsyncFileSinkProfile" : "INFO"
		},
		"/apsara/sls/ilogtail/status" : 
		{
			"AsyncFileSinkStatus" : "INFO"
		}
	},
	"Sinks" : 
	{
		"AsyncFileSink" : 
		{
			"Compress" : "",
                        "LogFilePath" : "/app/logs/ilogtail/ilogtail.LOG",
			"MaxDaysFromModify" : 300,
			"MaxLogFileNum" : 10,
			"MaxLogFileSize" : 20000000,
			"Type" : "AsyncFile"
		},
		"AsyncFileSinkProfile" : 
		{
			"Compress" : "",
			"LogFilePath" : "/app/logs/ilogtail/snapshot/ilogtail_profile.LOG",
			"MaxDaysFromModify" : 1,
			"MaxLogFileNum" : 61,
			"MaxLogFileSize" : 1,
			"Type" : "AsyncFile"
		},
		"AsyncFileSinkStatus" : 
		{
			"Compress" : "",
			"LogFilePath" : "/app/logs/ilogtail/snapshot/ilogtail_status.LOG",
			"MaxDaysFromModify" : 1,
			"MaxLogFileNum" : 61,
			"MaxLogFileSize" : 1,
			"Type" : "AsyncFile"
		}
	}
}

**Comment by leizhou666** [comment]
启动配置：
{
"cpu_usage_limit": 4,
"mem_usage_limit": 4096,
"max_bytes_per_sec": 209715200,
"process_thread_count": 8,
"send_request_concurrency": 40,
"buffer_file_num": 200,
"buffer_file_size": 104857600,
"buffer_file_path": "",
"max_read_buffer_size": 1048576,
"check_point_dump_interval": 300,
"check_point_filename": "/app/logs/ilogtail/checkpoint/logtail_check_point"
}

**Comment by leizhou666** [comment]
[2023-07-06-ilogtail.LOG](https://github.com/alibaba/ilogtail/files/11977214/2023-07-06-ilogtail.LOG)

**Comment by leizhou666** [comment]
还有一个额外的信息是logtail_check_point这个文件是打到了目录/app/logs/ilogtail/checkpoint，这个目录是挂载到node磁盘上的；
而docker_path_config.json这个文件是打到默认的目录/usr/local/ilogtail/checkpoint，这个目录没有挂载到node磁盘。
我不确定是不是由于这两个文件没有放到同一个目录导致的

**Reply by linrunqi08** [reply]
- name: ilogtail-log
          hostPath:
            path: /data/logs
        - name: ilog-agent-logs
          hostPath:
            path: /data/logs/ilogtail
            type: DirectoryOrCreate

这两个挂载有冲突吧

**Reply by linrunqi08** [reply]
你这里两个container，log-agent:v0.0.4这个是做什么的呢？

**Reply by leizhou666** [reply]
貌似是重复了，重复的挂载可能会导致checkpoint文件的这个问题吗

**Reply by leizhou666** [reply]
> 你这里两个container，log-agent:v0.0.4这个是做什么的呢？
log-agent是我们专门对业务pod监控的一个进程，监控pod的漂移，在对应的node上写入对应的ilogtail收集配置

**Reply by linrunqi08** [reply]
可以先把挂载拆分开，再看下吧

**Reply by leizhou666** [reply]
大佬，在使用过程中有其他用户反馈过这个情况吗？目前是我把多余的那个挂载删了，重启之后还有这个现象

**Reply by linrunqi08** [reply]
没有人反馈过，你现在的deployment再发一下？采集上来的数据有什么问题吗，比如采集重复？

**Reply by leizhou666** [reply]
好的，可能是我这边的问题，我先验证下

**Comment by leizhou666** [comment, marked as answer]
确认了，是我们自己的问题，我们起了两个ilogtail的daemonset，两个daemonset同时写导致的问题，抱歉。

**Reply by messixukejia** [reply]
能否把你们的采集配置在这里提交下？https://github.com/alibaba/ilogtail/tree/main/example_config/user_contrib

**Reply by leizhou666** [reply]
![image](https://github.com/alibaba/ilogtail/assets/97587488/9d479cee-6a8f-41c5-a7ad-9a96b3e3d199)
请问下我应该提交到哪里呢

**Reply by messixukejia** [reply]
https://github.com/alibaba/ilogtail/pull/1010/files
可以参考下这个，提交个新的文件即可。

**Reply by leizhou666** [reply]
大佬，问下你们需要这个文件用来做什么

**Reply by messixukejia** [reply]
就是将一些常用的配置进行归档，方便开发者借鉴使用

**Reply by leizhou666** [reply]
好的
